From 3d1618cfb48e75e01b469175036d2f28e8db47ac Mon Sep 17 00:00:00 2001
From: Joshua Ashton <joshua@froggi.es>
Date: Thu, 2 Nov 2023 03:01:19 +0000
Subject: [PATCH 38/41] drm/amd/display: Duplicate color management plane blobs
 correctly

This was done on the new copy rather the old one, so it was never actually duplicating any of this state.

Signed-off-by: Joshua Ashton <joshua@froggi.es>
---
 .../amd/display/amdgpu_dm/amdgpu_dm_plane.c   | 30 ++++++++++++-------
 1 file changed, 20 insertions(+), 10 deletions(-)

diff --git a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_plane.c b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_plane.c
index 9e3a33dcd..35d784474 100644
--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_plane.c
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_plane.c
@@ -1363,16 +1363,26 @@ static struct drm_plane_state *amdgpu_dm_plane_drm_plane_duplicate_state(struct
 		dc_plane_state_retain(dm_plane_state->dc_state);
 	}
 
-	if (dm_plane_state->degamma_lut)
-		drm_property_blob_get(dm_plane_state->degamma_lut);
-	if (dm_plane_state->ctm)
-		drm_property_blob_get(dm_plane_state->ctm);
-	if (dm_plane_state->shaper_lut)
-		drm_property_blob_get(dm_plane_state->shaper_lut);
-	if (dm_plane_state->lut3d)
-		drm_property_blob_get(dm_plane_state->lut3d);
-	if (dm_plane_state->blend_lut)
-		drm_property_blob_get(dm_plane_state->blend_lut);
+	if (old_dm_plane_state->degamma_lut) {
+		drm_property_blob_get(old_dm_plane_state->degamma_lut);
+		dm_plane_state->degamma_lut = old_dm_plane_state->degamma_lut;
+	}
+	if (old_dm_plane_state->ctm) {
+		drm_property_blob_get(old_dm_plane_state->ctm);
+		dm_plane_state->ctm = old_dm_plane_state->ctm;
+	}
+	if (old_dm_plane_state->shaper_lut) {
+		drm_property_blob_get(old_dm_plane_state->shaper_lut);
+		dm_plane_state->shaper_lut = old_dm_plane_state->shaper_lut;
+	}
+	if (old_dm_plane_state->lut3d) {
+		drm_property_blob_get(old_dm_plane_state->lut3d);
+		dm_plane_state->lut3d = old_dm_plane_state->lut3d;
+	}
+	if (old_dm_plane_state->blend_lut) {
+		drm_property_blob_get(old_dm_plane_state->blend_lut);
+		dm_plane_state->blend_lut = old_dm_plane_state->blend_lut;
+	}
 
 	dm_plane_state->degamma_tf = old_dm_plane_state->degamma_tf;
 	dm_plane_state->hdr_mult = old_dm_plane_state->hdr_mult;
-- 
2.43.0.232.ge79552d197

